### ISSUE TITLE
Improve how we handle third-party Ansible roles.

### ISSUE THREAD

**Issue by ndouglas** [department-of-veterans-affairs/va.gov-cms#8548]
Our use of BRD depends heavily on a number of third-party Ansible roles.  In many (most?) cases, these roles have been directly copied into the Devops codebase.  In some cases, these roles appear to have been directly copied into the codebase and the authoring information altered, which (among other complications) makes determining provenance difficult.  Sometimes, we open pull requests with upstream providers to reflect our local changes, but document the changes in a text file, which is haphazardly maintained from developer to developer, doesn't track upstream changes, etc.

This is not ideal.  We should use Ansible-Galaxy to manage our Ansible dependencies.  When we need to modify a task or other component, we should open an upstream PR or maintain a fork and reference it in `requirements.yml`, e.g.

```yaml
roles:
- 
  # Match the official name.
  name: datadog.datadog
  # URL to our fork.
  src: https://github.com/department-of-veterans-affairs/ansible-role-datadog.git
```

or 

```yaml
roles:
-
  # Match the official name.
  name: geerlingguy.apache
  # The name of our PR branch, commit, tag, etc.
  version: nd_retry_setup_for_flaky_repos
  # URL to the upstream role.
  src: https://github.com/geerlingguy/ansible-role-apache.git
```

This work is difficult to perform and test because of implicit assumptions made by each team individually and all teams collectively.  As the upstream and local copies diverge, this work becomes increasingly significant.  This should probably be done when and only when the BRD system becomes the sole province of Platform CMS, should that happen.

**Comment by ElijahLynn** [comment]
> This should probably be done when and only when the BRD system becomes the sole province of Platform CMS, should that happen.

Agree here. We had attempted to use Galaxy before but it was too big of a lift considering the amount of teams involved and the fact that we were not the core Platform Ops/Infra team, as it changes the development flow. I think also, there isn't a composer-patches equivalent yet for Ansible Galaxy, so patching a plugin has/had an unclear path, which we needed to do heavily at the time while we waiting for upstream PRs to be approved/merged or rejected.
